### PR TITLE
EDU-531 Get media duration

### PR DESCRIPTION
**Pull request by adrianaluchian**
https://educandu.atlassian.net/browse/EDU-531

### CODE REVIEW

File: src/plugins/interactive-media/interactive-media-editor.js [hunk 1]
```diff
@@ -1,102 +1,111 @@
...
 
 function InteractiveMediaEditor({ content, onContentChanged, publicStorage, privateStorage }) {
   const clientConfig = useService(ClientConfig);
+  const hiddenPlayerContainerRef = useRef(null);
   const { t } = useTranslation('interactiveMedia');
 
-  const { sourceType, sourceUrl, text, width, aspectRatio, showVideo } = content;
+  const getAspectRatioText = givenAspectRatio => `${givenAspectRatio.h}:${givenAspectRatio.v}`;
+  const ensureChaptersOrder = chapters => chapters.sort(by(chapter => chapter.startTimecode));
+  const getDefaultChapter = () => ({ key: uniqueId.create(), title: t('defaultChapterTitle'), startTimecode: 0 });
```
Comment (ahelmberger) [inline src/plugins/interactive-media/interactive-media-editor.js:41]: I think we should move this function into `interactive-media-info.js` (and also use it there when generating the whole default content).

File: src/plugins/interactive-media/interactive-media-editor.js [hunk 2]
```diff
@@ -1,102 +1,111 @@
...
+  const getMediaType = path => {
+    const sanitizedPath = (path || '').trim();
+    const extensionMatches = sanitizedPath.match(/\.([0-9a-z]+)$/i);
+    const extension = extensionMatches && extensionMatches[1];
+
+    if (!extension) {
+      return MEDIA_TYPE.none;
+    }
+    if (audioTypes.includes(extension)) {
+      return MEDIA_TYPE.audio;
```
Comment (ahelmberger) [inline src/plugins/interactive-media/interactive-media-editor.js:56]: I know this was there before, but I just realized we have (pretty much) the same code in `github-flavored-markdown.js`. Maybe you wanna check if we could easily unify those two places (there are a few more formats there, but they all should work)?

File: src/plugins/interactive-media/interactive-media-editor.js (hunk 1 above)
Comment (adrianaluchian) [reply src/plugins/interactive-media/interactive-media-editor.js]: Good idea, moved it there.

File: src/plugins/interactive-media/interactive-media-editor.js (hunk 2 above)
Comment (adrianaluchian) [reply src/plugins/interactive-media/interactive-media-editor.js]: Nice, I didn't know about this other place, thanks 👍 
Unified the media type identification.

File: src/utils/media-utils.js [hunk 3]
```diff
@@ -0,0 +1,21 @@
+import { MEDIA_TYPE } from '../domain/constants.js';
+
+const audioUrlExtensions = ['aac', 'm4a', 'mp3', 'oga', 'ogg', 'wav', 'flac'];
+const videoUrlExtensions = ['mp4', 'm4v', 'ogv', 'webm', 'mpg', 'mpeg', 'mov', 'avi', 'mkv'];
+
+export const getMediaType = url => {
+  const sanitizedUrl = (url || '').trim();
+  const extensionMatches = sanitizedUrl.match(/\.([0-9a-z]+)$/i);
+  const extension = extensionMatches && extensionMatches[1];
```
Comment (ahelmberger) [inline src/utils/media-utils.js:9]: Just to be on the safe side I'd propose `const extension = extensionMatches?.[1]?.toLowerCase();` realizing that some applications produce files like `MYSOUND.MP3`

Comment (adrianaluchian) [reply src/utils/media-utils.js]: Oh of course, the regex that was in the markdown plugin was case insensitive 🤦‍♀️ 
Fixed, thanks for noticing it!

**Review by ahelmberger — APPROVED** [review]

**Pull request merged**